owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/CreateRemoteShareOperation.kt

```diff
@@ -0,0 +1,191 @@
+                .add(PARAM_SHARE_WITH, shareWith)
+
+            if (name.isNotEmpty()) {
+                formBodyBuilder.add(PARAM_NAME, name)
+            }
+
+            if (expirationDateInMillis > INITIAL_EXPIRATION_DATE_IN_MILLIS) {
+                val dateFormat = SimpleDateFormat(FORMAT_EXPIRATION_DATE, Locale.getDefault())
+                val expirationDate = Calendar.getInstance()
+                expirationDate.timeInMillis = expirationDateInMillis
+                val formattedExpirationDate = dateFormat.format(expirationDate.time)
+                formBodyBuilder.add(PARAM_EXPIRATION_DATE, formattedExpirationDate)
+            }
+
+            if (publicUpload == true) {
+                formBodyBuilder.add(PARAM_PUBLIC_UPLOAD, publicUpload.toString())
+            }
+            if (!password.isNullOrEmpty()) {
+                formBodyBuilder.add(PARAM_PASSWORD, password!!)
+            }
+            if (RemoteShare.DEFAULT_PERMISSION != permissions) {
+                formBodyBuilder.add(PARAM_PERMISSIONS, Integer.toString(permissions))
+            }
+
+            val requestUri = client.baseUri
+            val uriBuilder = requestUri.buildUpon()
+            uriBuilder.appendEncodedPath(ShareUtils.SHARING_API_PATH)
+
+            val postMethod = PostMethod(URL(uriBuilder.build().toString()))
+
+            postMethod.setRequestBody(formBodyBuilder.build())
+
+            postMethod.setRequestHeader(HttpConstants.CONTENT_TYPE_HEADER, HttpConstants.CONTENT_TYPE_URLENCODED_UTF8)
+            postMethod.addRequestHeader(RemoteOperation.OCS_API_HEADER, RemoteOperation.OCS_API_HEADER_VALUE)
+
+            val status = client.executeHttpMethod(postMethod)
+
+            val parser = ShareToRemoteOperationResultParser(
+                ShareXMLParser()
+            )
+
+            if (isSuccess(status)) {
+                parser.oneOrMoreSharesRequired = true
+                parser.ownCloudVersion = client.ownCloudVersion
+                parser.serverBaseUri = client.baseUri
+                result = parser.parse(postMethod.responseBodyAsString)
+
+                if (result.isSuccess && getShareDetails) {
+
+                    // TODO Use executeHttpMethod
+                    // retrieve more info - POST only returns the index of the new share
+                    val emptyShare = result.data.shares[0]
+                    val getInfo = GetRemoteShareOperation(
+                        emptyShare.remoteId
+                    )
+                    result = getInfo.execute(client)
+                }
+
+            } else {
+                result = parser.parse(postMethod.responseBodyAsString)
+            }
+
+        } catch (e: Exception) {
+            result = RemoteOperationResult(e)
+            Log_OC.e(TAG, "Exception while Creating New Share", e)
+        }
+
+        return result
+    }
+
+    private fun isSuccess(status: Int): Boolean {
```

Comment:
it can be refactored to 

`private fun isSuccess(status: Int): Boolean = status == HttpConstants.HTTP_OK`

Reply:
private fun isSuccess(status: Int): Boolean = status == HttpConstants.HTTP_OK
@hannesa2
@jcmontesmartos
jcmontesmartos reviewed 10 days ago
owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/RemoteShare.kt Outdated

    constructor(path: String?) {
        resetData()
        if (path == null || path.isEmpty() || !path.startsWith(FileUtils.PATH_SEPARATOR)) {
@jcmontesmartos
jcmontesmartos 10 days ago

if (path.isNullOrEmpty() || !path.startsWith(FileUtils.PATH_SEPARATOR))
@hannesa2
@jcmontesmartos
jcmontesmartos reviewed 10 days ago
owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/RemoteShare.kt
        expirationDate = 0
        sharedWithAdditionalInfo = ""
        isFolder = false
        userId = -1
@jcmontesmartos
jcmontesmartos 10 days ago

why don't you use constants for initial values?
@davigonz
davigonz 10 days ago Author Member

This is a question that should have been done in a code review 5 years ago:

https://github.com/owncloud/android-library/blame/87a8b418ded714e64aa77c9a007801b6cf4a985e/owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/OCShare.java#L132

Not sure if is going to be possible (in terms of time) to improve the new code added by me and legacy code (converted from Java to Kotlin). Being strict, just the new code should be reviewed and changed in code reviews.

Otherwise...

17201963
@hannesa2
@jcmontesmartos
jcmontesmartos reviewed 10 days ago
owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/RemoteShare.kt
        permissions = source.readInt()
        sharedDate = source.readLong()
        expirationDate = source.readLong()
        isFolder = source.readInt() == 0
@jcmontesmartos
jcmontesmartos 10 days ago

0 ?? hardcoded??
@hannesa2
@jcmontesmartos
jcmontesmartos reviewed 10 days ago
owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/RemoteShare.kt Outdated
        remoteId = source.readLong()
    }

    override fun describeContents(): Int {
@jcmontesmartos
jcmontesmartos 10 days ago

override fun describeContents(): Int = this.hashCode()
@hannesa2
@jcmontesmartos
jcmontesmartos reviewed 10 days ago
owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/RemoteShare.kt Outdated
        dest.writeString(shareLink)
        dest.writeLong(fileSource)
        dest.writeLong(itemSource)
        dest.writeString(if (shareType == null) "" else shareType!!.name)
@jcmontesmartos
jcmontesmartos 10 days ago

replace that conditional by an elvis operator like this

shareType?.name ?: ""
@hannesa2
@jcmontesmartos
jcmontesmartos reviewed 10 days ago
...ibrary/src/main/java/com/owncloud/android/lib/resources/shares/RemoveRemoteShareOperation.kt Outdated
 *
 * @param remoteShareId Share ID
 */
    (private val mRemoteShareId: Long) : RemoteOperation<ShareParserResult>() {
@jcmontesmartos
jcmontesmartos 10 days ago

converted file?? you have only one constuctor and java doc should be above class declaration
@davigonz
davigonz 10 days ago Author Member

How can I include java doc for a constructor if is part of the class declaration in Kotlin instead of a separate method? 🤦‍♂️
@davigonz
davigonz 10 days ago Author Member

I can put it above but is pretty ugly

screenshot 2019-03-05 at 17 40 05
@hannesa2
@jcmontesmartos
jcmontesmartos reviewed 10 days ago
...ibrary/src/main/java/com/owncloud/android/lib/resources/shares/RemoveRemoteShareOperation.kt Outdated
 *
 * @param remoteShareId Share ID
 */
    (private val mRemoteShareId: Long) : RemoteOperation<ShareParserResult>() {
@jcmontesmartos
jcmontesmartos 10 days ago

BTW, I thought you were removing m prefix due to it is not a java file 😄
@hannesa2
@jcmontesmartos
jcmontesmartos reviewed 10 days ago
...omLibrary/src/main/java/com/owncloud/android/lib/resources/shares/SharePermissionsBuilder.kt Outdated
     */
    private int mPermissions = OCShare.READ_PERMISSION_FLAG;    // READ is minimum permission
    /** Set of permissions  */
    private var mPermissions = RemoteShare.READ_PERMISSION_FLAG    // READ is minimum permission
@jcmontesmartos
jcmontesmartos 10 days ago

the same comment as before
@hannesa2
@jcmontesmartos
jcmontesmartos reviewed 10 days ago
...omLibrary/src/main/java/com/owncloud/android/lib/resources/shares/SharePermissionsBuilder.kt Outdated
@@ -101,7 +99,7 @@ private void updatePermission(int permissionsFlag, boolean enable) {
     *
     * @return An int value representing the accumulated set of permissions.
     */
    public int build() {
        return mPermissions;
    fun build(): Int {
@jcmontesmartos
jcmontesmartos 10 days ago

fun build(): Int = permissions
@hannesa2
@jcmontesmartos
jcmontesmartos reviewed 10 days ago
...ibrary/src/main/java/com/owncloud/android/lib/resources/shares/UpdateRemoteShareOperation.kt Outdated
    private var name: String? = null

    init {
        password = null               // no update
@jcmontesmartos
jcmontesmartos 10 days ago

hardcoded values....
@davigonz
davigonz 10 days ago Author Member

Creating constants for null values would not be too much?
@hannesa2
@jcmontesmartos
jcmontesmartos reviewed 10 days ago
...ibrary/src/main/java/com/owncloud/android/lib/resources/shares/UpdateRemoteShareOperation.kt Outdated
     * Empty string clears the current name.
     * Null results in no update applied to the name.
     */
    fun setName(name: String) {
@jcmontesmartos
jcmontesmartos 10 days ago

why don't you take advantage of kotlin language such as getter and setters as you have already did in the last commit?
@hannesa2
@jcmontesmartos
jcmontesmartos reviewed 10 days ago
...ibrary/src/main/java/com/owncloud/android/lib/resources/shares/UpdateRemoteShareOperation.kt Outdated
        return result
    }

    private fun isSuccess(status: Int): Boolean {
@jcmontesmartos
jcmontesmartos 10 days ago

private fun isSuccess(status: Int): Boolean = status == HttpConstants.HTTP_OK
@hannesa2
@davigonz davigonz force-pushed the new_arch/get_public_shares_for_file branch from f5732ac to 30bfa4e 9 days ago
@jcmontesmartos
jcmontesmartos approved these changes 9 days ago
jcmontesmartos left a comment

JC's approval!!
@davigonz
Member Author
davigonz commented 9 days ago

    JC's approval!!

Thanks a lot
@davigonz davigonz force-pushed the new_arch/get_public_shares_for_file branch from 30bfa4e to ba47024 8 days ago
davigonz added some commits on Jan 3
@davigonz
Rename OCShare to RemoteShare
5276cb8
@davigonz
Use different type for remoteId
cc14ae6
@davigonz
Make OwnCloudVersion parcelable
4e30077
@davigonz
Fix conflicts with master
34220c2
@davigonz
Convert used classes to Kotlin
d19160b
@davigonz
Use openfortesting for new Kotlin files needed in tests
2d65ea5
@davigonz
Fix tiny bug when parsing shares
6b160a3
@davigonz
Fix some dependencies
85e23f7
@davigonz
Fix build
7b5e00f
@davigonz
Apply code review changes
36d2b50
@davigonz
2nd code review changes
8823e2a
@davigonz davigonz force-pushed the new_arch/get_public_shares_for_file branch from ba47024 to 8823e2a 2 days ago
@davigonz davigonz changed the base branch from master to new_arch/shares 2 days ago
Merge state

Add more commits by pushing to the new_arch/get_public_shares_for_file branch on owncloud/android-library.
Review requested
Review has been requested on this pull request. It is not required to merge.
@abelgardep
abelgardep was requested for review
All checks have passed
5 successful checks
continuous-integration/travis-ci/pr — The Travis CI build passed
Details
license/cla — Contributor License Agreement is signed.
Details
security/snyk - build.gradle (owncloud) — No new issues
Details
security/snyk - pom.xml (owncloud) — No new issues
Details
security/snyk - sample_client/build.gradle (owncloud) — No manifest changes detected


it can be refactored to

`private fun isSuccess(status: Int) = status == HttpConstants.HTTP_OK`